quick_select/quick_select.c

```diff
@@ -0,0 +1,46 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int partition(int *ar, int left, int right, int pivot_idx) {
+   int pivot_value = ar[pivot_idx];
+   int temp = ar[right];
+   ar[right] = ar[pivot_idx];
+   ar[pivot_idx] = temp;
+
+   int store_idx = left;
+   while (left < right) {
+      if (ar[left] < pivot_value) {
+         temp = ar[store_idx];
+         ar[store_idx] = ar[left];
+         ar[left] = temp;
+         store_idx++;
+      }
+      left++;
+   }
+   temp = ar[right];
+   ar[right] = ar[store_idx];
+   ar[store_idx] = temp;
+   return store_idx;
+}
+
+int quick_select(int *ar, int left, int right, int pos) {
+   int pivot_index = 0;
+   if (left == right)
+      return ar[left];
+   pivot_index = right - 1;
+   pivot_index = partition(ar, left, right, pivot_index);
+   if (pos == pivot_index) {
+      return ar[pivot_index];
+   } else if (pos < pivot_index) {
+      return quick_select(ar, left, pivot_index - 1, pos);
+   } else {
+      return quick_select(ar, pivot_index + 1, right, pos);
+   }
+   return 0;
```

Comment:
Sir it is because of a warning that all the return statements are inside the conditional blocks and there is a warning issued at this point because function will not return any value if in any hypothetical case it doesn't go into any conditional block statement, and it requires a return statement.